src/storage/FixedConvertingStore.ts

```diff
@@ -0,0 +1,45 @@
+import type { Representation } from '../ldp/representation/Representation';
+import type { RepresentationPreference } from '../ldp/representation/RepresentationPreference';
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import type { Conditions } from './Conditions';
+import type { RepresentationConverter } from './conversion/RepresentationConverter';
+import { PassthroughStore } from './PassthroughStore';
+import type { ResourceStore } from './ResourceStore';
+
+/**
+ * Store that converts incoming data when required.
+ * If the content-type of an incoming representation does not match one of the stored types it will be converted.
+ */
+export class FixedConvertingStore extends PassthroughStore {
+  private readonly types: string[];
+  private readonly converter: RepresentationConverter;
+
+  public constructor(source: ResourceStore, converter: RepresentationConverter, types: string[]) {
+    super(source);
+    this.converter = converter;
+    this.types = types;
+  }
+
+  public async addResource(container: ResourceIdentifier, representation: Representation,
+    conditions?: Conditions): Promise<ResourceIdentifier> {
+    // We can potentially run into problems here if we convert a turtle document where the base IRI is required,
```

Comment:
I would create an issue for this once the PR is merged.

Reply:
Good idea, will probably stay there for a while because I have no idea how to cleanly solve this 😄